rimage/transform/parallel_projection_test.go

```diff
@@ -0,0 +1,225 @@
+		pose := spatialmath.NewPose(r3.Vector{X: 0, Y: 0, Z: 0}, spatialmath.NewOrientationVector())
+		ppRM := NewParallelProjectionOntoXZWithRobotMarker(&pose)
+
+		pointcloud := pc.New()
+		d := pc.NewBasicData()
+		p1 := r3.Vector{X: -2, Y: 10, Z: -3}
+		err := pointcloud.Set(p1, d)
+		test.That(t, err, test.ShouldBeNil)
+		p2 := r3.Vector{X: 10, Y: 10, Z: 10}
+		err = pointcloud.Set(p2, d)
+		test.That(t, err, test.ShouldBeNil)
+
+		im, unusedDepthMap, err := ppRM.PointCloudToRGBD(pointcloud)
+		test.That(t, err, test.ShouldBeNil)
+		test.That(t, im.Width(), test.ShouldEqual, imageWidth)
+		test.That(t, im.Height(), test.ShouldEqual, imageHeight)
+		test.That(t, unusedDepthMap, test.ShouldBeNil)
+
+		minX := math.Min(math.Min(pose.Point().X, p1.X), p2.X)
+		maxX := math.Max(math.Max(pose.Point().X, p1.X), p2.X)
+		minZ := math.Min(math.Min(pose.Point().Z, p1.Z), p2.Z)
+		maxZ := math.Max(math.Max(pose.Point().Z, p1.Z), p2.Z)
+
+		scaleFactor := math.Min((imageWidth-1)/(maxX-minX), (imageHeight-1)/(maxZ-minZ))
+
+		robotMarkerExpectedPos := image.Point{
+			X: int(math.Round((pose.Point().X - minX) * scaleFactor)),
+			Y: int(math.Round((pose.Point().Z - minZ) * scaleFactor)),
+		}
+
+		colorAtPos := im.GetXY(robotMarkerExpectedPos.X, robotMarkerExpectedPos.Y)
+		expectedRobotMarkerColor := rimage.NewColor(255, 0, 0)
+		test.That(t, colorAtPos, test.ShouldResemble, expectedRobotMarkerColor)
+
+		point1ExpectedPos := image.Point{
+			X: int(math.Round((p1.X - minX) / scaleFactor)),
+			Y: int(math.Round((p1.Z - minZ) / scaleFactor)),
+		}
+
+		colorAtPoint1 := im.GetXY(point1ExpectedPos.X, point1ExpectedPos.Y)
+		expectedPoint1Color, err := getColorFromProbabilityValue(d)
+		test.That(t, err, test.ShouldBeNil)
+		test.That(t, colorAtPoint1, test.ShouldResemble, expectedPoint1Color)
+
+		point2ExpectedPos := image.Point{
+			X: int(math.Round((p2.X - minX) / scaleFactor)),
+			Y: int(math.Round((p2.Z - minZ) / scaleFactor)),
+		}
+
+		colorAtPoint2 := im.GetXY(point2ExpectedPos.X, point2ExpectedPos.Y)
+		expectedPoint2Color, err := getColorFromProbabilityValue(d)
+		test.That(t, err, test.ShouldBeNil)
+		test.That(t, colorAtPoint2, test.ShouldResemble, expectedPoint2Color)
+	})
+
+	t.Run("Project an imported pointcloud", func(t *testing.T) {
+		p := spatialmath.NewPose(r3.Vector{X: 0, Y: 0, Z: 0}, spatialmath.NewOrientationVector())
+		ppRM := NewParallelProjectionOntoXZWithRobotMarker(&p)
+
+		pcdFile, err := os.Open(artifact.MustPath("pointcloud/test_short.pcd"))
+		test.That(t, err, test.ShouldBeNil)
+
+		PC, err := pc.ReadPCD(pcdFile)
+		test.That(t, err, test.ShouldBeNil)
+
+		im, unusedDepthMap, err := ppRM.PointCloudToRGBD(PC)
+		test.That(t, err, test.ShouldBeNil)
+		test.That(t, im.Width(), test.ShouldEqual, imageWidth)
+		test.That(t, im.Height(), test.ShouldEqual, imageHeight)
+		test.That(t, unusedDepthMap, test.ShouldBeNil)
+	})
+
+	t.Run("Test that projecting two offset pointclouds will produce same image", func(t *testing.T) {
```

Comment:
Can we please add the following tests:
1. Behavior when robot marker falls outside of standard deviation.
2. Behavior when either maxX==minX, maxZ==minZ, or both.
3. Behavior when any of the point cloud points have nil data